src/memray/_memray/record_writer.h

```diff
@@ -58,47 +87,166 @@ bool inline RecordWriter::writeString(const char* the_string)
     return d_sink->writeAll(the_string, strlen(the_string) + 1);
 }
 
+bool inline RecordWriter::writeVarint(size_t rest)
+{
+    unsigned char next_7_bits = rest & 0x7f;
```

Comment:
Discussed offline - `ULL` isn't really quite right, since that's too large for a `size_t` on 32-bit. I used `0x7fULL` in the decoder, and Pablo was proposing it here in the encoder for symmetry. Instead, I'll update the reader to do a `static_cast<size_t>` in the case where it needs to be promoted.